### ISSUE TITLE
embenchen zlib test failure

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#152]
This was caused by https://codereview.chromium.org/1205353002 / 35eb3a0260, 'turbofan] Optimize BooleanNot conditions to Branch nodes.'

Preliminary investigation shows that benchmarks lib does not fail, but this asm test does. 

I am working to get a reduced test case. Some testing shows that this lowering is commonly used in many places, many of the simile branch & loops tests in test-run-machops use this case, but it does not fail in those cases.

**Comment by paul99** [comment]
This is now on the skip list with https://codereview.chromium.org/1209253005

Testing has showed that many simple tests, as well as most of the embenchen tests use the kBooleanNot clause in CommonOperatorReducer::ReduceBranch. All tests pass except zlib.

zlib, and the other embenchen tests I checked use this optimization _many_ times. For example, it is used 66 times in the _malloc() function alone (which is in the common part of the embenchen toolchain - like a standard lib.) I think the naming convention is leading '_' for these lib functions, and it's likely true that the libs are OK, since other tests pass.

My plan is to individually select functions to optimize with TF, using --turbo-filter, and iterate thru zlib , TF-compiling only one function at a time, to identify which TF-compiled function leads to the error. Then I can start looking at TF code-generation for that single function.

However, my first cut today with --turbo-filter was _not_ limiting to a single TF-compiled function. There is a bit more complexity to this option than I thought, so I am investigating that now.

**Comment by paul99** [comment]
Fixed by Dusan in https://codereview.chromium.org/1232313007, and removed zlib from skip list.

Landed as https://crrev.com/82339644b24eccad008335a8bcbcfcd6532823f0

This needs to be back-merged to 4.5, which I will do. Issue stays open until complete.

**Comment by paul99** [comment]
Merged to 4.5. Closing.

```
commit de9a9c2b55e04cca0a7144adf4122dd47a0683d2
Author: Paul Lind <paul.lind@imgtec.com>
Date:   Tue Jul 21 07:01:51 2015 -0700

    Version 4.5.103.10 (cherry-pick)

    Merged 82339644b24eccad008335a8bcbcfcd6532823f0

    MIPS: Fix missing fallthru handling for some branch cases in TF.
```
